File: Problems/Algorithms/27. Remove Element/RemoveElement.java
class Solution {
    public int removeElement(int[] nums, int val) {
        int ans = nums.length;
        int i = 0;
        while (i < ans) {
            if (val == nums[i]) {
                int temp = nums[i];
                nums[i] = nums[ans-1];
                nums[ans-1] = temp;
                ans--;
            } else {
                i++;
            }
        }
        
        return ans;
    }
}

File: Explore/Learn/Introduction to Data Structure/Arrays 101/Remove Element/RemoveElement.kt
class Solution {
    fun removeElement(nums: IntArray, `val`: Int): Int {
        val length = nums.size
        var i = 0
        var j = length - 1
        
        while (i <= j) {
            if (nums[i] == `val`) {
                var tmp = nums[i]
                nums[i] = nums[j]
                nums[j] = tmp
                j--
            } else {
                i++    
            }
        }
        
        return j+1
    }
}

